### DISCUSSION TITLE
CW transmitting not perfect

### DISCUSSION THREAD

**Discussion by rosmont57** [gdyuldin/x6100_gui#24] [Q&A]
Hello..I noticed that, unlike the cw decoding, which works very well, the cw encoding both with the key and with a stored message is not precise. Some letters every now and then are not encoded perfectly. I also tried to modify all the parameters related to the cw but I did not solve it. However, I noticed that the same problem exists even using the original xiegu firmware. It is a bit annoying even if not of vital importance.Has anyone encountered the same problem? Thanks

**Comment by gdyuldin** [comment]
Well, CW encoding from the key is performed within BASE board - GUI application can't control it, as far as I know.
Regarding stored message - I should check how it passed to the BASE, perhaps, something might be improved here.

**Reply by rosmont57** [reply]
Thank you..! Congratulations for your great work. You have improved the potential of the Xiegu 6100 in a huge way!! A kitten is becoming a tiger!! thanks again and good work!

**Comment by rikmac** [comment]
Hello, I am Riccardo IW5DGQ. In CW mode, it's not possible to use the Iambic key, at least on my end, because occasionally the 'dit'  are not sent, creating errors, whereas with the original software this doesn't happen. Could you please do some tests to verify and see if it can be fixed?

Thank you for the work you are doing and sharing with us radio enthusiasts.

**Reply by gdyuldin** [reply]
Hi Riccardo. I still learning CW and only using x6100 in a training mode, but it's working fine (both OEM and R1CBU). All work with hardware key (straight or iambic) happens within BASE part of x6100, which uses its own independent logic, which remains the same for OEM and alternative GUI.

**Reply by rikmac** [reply]
Hi, thanks for the reply, I also encountered the problem using the flrig program with both the original and r1cbu, perhaps it will be a connection or cable problem, I will do other tests as soon as possible, greetings.
